Dockerfile

```diff
@@ -0,0 +1,166 @@
+ARG unitycatalog_uid=185
+ARG unitycatalog_home="/opt/unitycatalog"
+ARG unitycatalog_repo="${unitycatalog_home}/repo"
+ARG unitycatalog_jar="server/target"
+ARG unitycatalog_etc="etc"
+ARG unitycatalog_bin="bin"
+ARG unitycatalog_user_name="unitycatalog"
+ARG unitycatalog_user_home="home"
+ARG unitycatalog_user_basedir="${unitycatalog_home}/${unitycatalog_user_home}"
+# Specify any custom parameters necessary to generate
+# the Uber-Jar by SBT. 
+# Note: The default allocated heam memory size is too small
+# and will cause the process to fail when attempting to compile
+# and generate the Uber-Jar. Therefore it is important to choose
+# a size large enough for the compiler to run. 
+ARG sbt_args="-J-Xmx5G"
+
+
+
+# ###### STAGE 1 ###### #
+# Building the Uber-Jar #
+#########################
+
+FROM eclipse-temurin:22-jdk-alpine AS build_uc
+
+ARG unitycatalog_repo
+ARG sbt_args
+
+# Install any required packages
+RUN <<EOF
+    set -ex;
+    apk update;
+    apk upgrade; 
+    apk add bash git;
+    rm -R /var/cache/apk/*;
+EOF
+
+# Create temporary directories for initialization files
+RUN mkdir -p "${unitycatalog_repo}";
+
+WORKDIR "${unitycatalog_repo}"
+
+COPY . .
+
+RUN build/sbt ${sbt_args} assembly
```

Comment:
For now we're focussing on the server component, so maybe only create a JAR for that project:

```suggestion
RUN build/sbt ${sbt_args} server/assembly
```

I noticed that it takes quite a bit of time, WDYT?

Reply:
Great spot! I was going to ask you actually if we can build only the server classes without the cli.
I've added your suggestion and rebuilt, and it's much faster now 😍